src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingMiner.java

```diff
@@ -649,11 +664,13 @@ public void deserialize(@NotNull final ByteBuf buf)
             super.deserialize(buf);
             current = buf.readInt();
             final int size = buf.readInt();
-            levels = new int[size];
+            nodesOfLevels = new int[size];
+            yDepthOfLevels = new int[size];
 
             for (int i = 0; i < size; i++)
             {
-                levels[i] = buf.readInt();
+                nodesOfLevels[i] = buf.readInt();
```

Comment:
Make it an array of Tuple<Integer, Integer> no need for two arrays.

Reply:
Sure